Revised index.rst and leading numbers on RTD directories

CodingApps opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied the relevant labels to this PR
- [ ] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

<!--- For the title, please observe the following rules:
	- Provide a concise and self-descriptive title
	- Do not include the applicable issue number in the title, do it in the PR body
	- If the PR is not ready for review, convert it to a draft.
-->

## Description
Revised index.rst to have the RTD doc directory names without additional numbers ahead of the names.

Directories had leading numbers also taken off, are now :
  overview/
  getting_started/
  contributing/
  about/
  other-resources/hardware/
  cli_reference/
  api_reference/

Existing tests for RTD docs should cover these links. 

## Linked issues
Submitting to review by : @jcohenadad 
Resolves : https://github.com/shimming-toolbox/shimming-toolbox/issues/198 "Remove numbers in URLs from RTD"

CodingApps committed:
Revised index.rst and leading numbers on RTD directories

Revised index.rst to have the RTD doc directory names without additional numbers ahead of the names.

Directories had leading numbers also taken off, are now :
  overview/
  getting_started/
  contributing/
  about/
  other-resources/hardware/
  cli_reference/
  api_reference/

po09i reviewed (commented):
Thanks for your contribution!

There are numbers left in the file names for example:
- shimming-toolbox/docs/source/getting_started/1_installation.rst

We should fix this but preserve the order in the TOC.

CodingApps committed:
Revised filenames in /getting_started and /overview to remove numbers

Revised filenames in /getting_started and /overview to remove numbers, also updated TOC.

In TOC perserved order, for example :

getting_started/installation.rst
getting_started/getting_started.rst
getting_started/help.rst

CodingApps commented:
@po09i Thanks for checking this! 

Revised filenames in /getting_started and /overview to remove numbers, also updated TOC. Pushed it to the PR. 

In TOC perserved order, folder sample :

getting_started/installation.rst
getting_started/getting_started.rst
getting_started/help.rst

> Thanks for your contribution!
> 
> There are numbers left in the file names for example:
> 
> * shimming-toolbox/docs/source/getting_started/1_installation.rst
> 
> We should fix this but preserve the order in the TOC.

po09i reviewed (approved):
LGTM 👍
